clang/lib/Driver/Driver.cpp

```diff
@@ -3575,7 +3575,7 @@ class OffloadingActionBuilder final {
       Arg *SYCLLinkTargets = Args.getLastArg(
                                   options::OPT_fsycl_link_targets_EQ);
       WrapDeviceOnlyBinary = Args.hasArg(options::OPT_fsycl_link_EQ);
-      DeviceCodeSplit = Args.hasArg(options::OPT_fsycl_device_code_split);
+      DeviceCodeSplit = Args.hasArg(options::OPT_fsycl_device_code_split_EQ);
```

Comment:
```suggestion
      DeviceCodeSplit = Args.getLastArg(options::OPT_fsycl_device_code_split_EQ);
```

Reply:
Done.

Reply:
`getLastArg()` yeilds `Arg *`, which is then implicitly converted to `bool`. If the aim is to check for `-fsycl-device-code-split=` value, `DeviceCodeSplit`'s type had better be turned into a string representation, which can be retrieved by `Arg::getValue()`, or set to `""` if `getLastArg()` yields `nullptr`.

Reply:
`DeviceCodeSplit` variable is needed here only to check presence of device code split option. Value is checked in SYCL toolchain. I think I can try implement request from @bader and add `none` value for `-fsycl-device-code-split` option, then conversion `DeviceCodeSplit`'s type to a string representation will make sense.

Reply:
Looks reasonable. There's also an option to leave `DeviceCodeSplit` boolean and set it like this:
```
auto *DeviceCodeSplitArg = Args.getLastArg(OPT_fsycl_device_code_split_EQ);
DeviceCodeSplit = DeviceCodeSplitArg && DeviceCodeSplitArg->getValue() != "none";
```

Reply:
Done.

Reply:
I understand that no additional handling of `none` is needed at the `SYCLToolchain` level, but could you please confirm this?

Reply:
By what?

Reply:
By a comment akin to "Yes, I've checked myself and `none` doesn't need to be handled there", or by a new commit that does take it into account... Apparently :)

Reply:
Yes, I've checked myself :)